### ISSUE TITLE
Smart Contract List

### ISSUE THREAD

**Issue by janewang** [stellar/laboratory#1196]
### What problem does your feature solve?
Create Smart Contract List page

![Image](https://github.com/user-attachments/assets/0a810b0e-25bb-4b9f-8650-6d6c21fe96ff) https://www.figma.com/design/yuikaFOPmJifyAfS90ER6T/Laboratory?node-id=6277-29435&m=dev

**Comment by nagxsan** [comment]
Can I please work on this issue?

**Comment by leighmcculloch** [comment]
What’s the user story for this feature?

Is this a UI for a developer to find specific contracts? or contracts that are related or used together? or to discover new contracts?

Contracts are unbounded, like accounts, Im wondering how practical for the user to scroll through an endless list of contracts. 

Will there be the ability to search, filter, and sort?

**Comment by janewang** [comment]
@leighmcculloch I believe we are holding off on search, filter, and sort features on this screen until there's a decision on 1) where to get this data and 2) if there exists a backend. To your point, I think it makes a lot of sense to be able to search, and eventually filter and sort.

cc @sagpatil @quietbits

**Comment by leighmcculloch** [comment]
> What’s the user story for this feature?

👆🏻 

> we are holding off on search, filter, and sort features on this screen until there's a decision on 1) where to get this data

I think the user story will define which features are required, and can then influence which data source is selected to ensure the data source selected can provide the features needed for the user story.

**Comment by janewang** [comment]
We want to provide a dashboard displaying mainnet contracts. Initially, this will include a comprehensive list of all contracts on the mainnet. Over time, we may refine this to highlight only high-activity (hot) contracts. We want the contract link to be clickable such that it could take the user to the contract details page and load the details related to this contract. 

The data source will be determined by the engineering team. @sagpatil has proposed reaching out to the data team to explore potential sources for this data.

**Comment by leighmcculloch** [comment]
I'm still unclear on the user story. It sounds like this feature, exploring network state, is a feature of a block explorer rather than a developer tool like the lab. Why would a developer come to the lab to explore network state than a block explorer?

**Comment by janewang** [comment]
I don't think it categorically make sense to say: does this feature provide network state? If so, we remove it. I also don't think it is categorically true to say that product A provides a feature, therefore product B cannot provide this. All should be use case based. 

For example, Lab already offers the RPC and Horizon API endpoints and we make those endpoints callable. If you think that it is the responsibility of the block explorer to provide network state and it's the responsibilities for Horizon and RPC to provide APIs,, then we may as well remove API explorer or make the API not callable, just copy and paste as some command line and not use the Lab to make the calls and provide responses (which include the current network state).  The developers can make command line calls then paste XDR into Lab. 

In this case, we are building a contract explorer. This is where the developer starts with:

<img width="1359" alt="Image" src="https://github.com/user-attachments/assets/20ff95d1-c624-4a89-bccf-bfd96bc01092" />

> Why would a developer come to the lab to explore network state than a block explorer?

Instead of the current screen, where the developer has to locate the contract elsewhere, the developer would begin with a list of contracts. They can search these contracts by metadata, allowing them to view the contract details. From there, they can download the WASM file and deploy it locally, read the code, or attempt to invoke the contract, etc. 

I assure you if you give new developer the contract explorer page and ask them to figure out what to put into the c-name, it would be not obvious.

**Comment by leighmcculloch** [comment]
> If you think that it is the responsibility of the block explorer to provide network state and it's the responsibilities for Horizon and RPC to provide APIs,, then we may as well remove API explorer

The lab provides an API explorer primarily so that devs can learn and explore the API, not to replace the use of the SDKs or CLI.

> Instead of the current screen, where the developer has to locate the contract elsewhere, the developer would begin with a list of contracts.

Instead of building out a large amount of functionality to support listing all contracts, pagination, search, filtering, which I think bridges into the responsibility of an block explorer, could we instead list just one or two example contract IDs? For example, we could list just the native (XLM) asset.

The list will be too long to meaningfully find anything else is my concern, and it's a non-trivial amount of effort to build this and that effort could be directed elsewhere.

**Comment by janewang** [comment]
> it's a non-trivial amount of effort to build this and that effort could be directed elsewhere.

We are going to follow the API provided by Orbit lens. As it is (without adding additional feature like search and filter) according to the engineers, this is not hard.
